### PR TITLE
PP-4886: Tests for invalid metadata

### PR DESCRIPTION
**Pull request by oswaldquek**
Validations happen in RequestJsonParser, which has become rather messy; but
this will be refactored in a separate PR.

### CODE REVIEW

File: src/test/java/uk/gov/pay/api/it/validation/PaymentResourceMetadataValidationFailuresITest.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        
+        Map<String, Object> metadata = new HashMap<>() {{ put("key", value); }};
+        var createChargeRequestParams = createChargeRequestParamsBuilder.withMetadata(metadata).build();
+        assertMetadataValidationError(createChargeRequestParams, "Invalid attribute value: " + expectedMessage);
+    }
+
+    @Test
+    public void moreThan10Keys() {
+        
+        Map<String, Object> metadata = new HashMap<>() {{
```
Comment (danworth) [inline src/test/java/uk/gov/pay/api/it/validation/PaymentResourceMetadataValidationFailuresITest.java]: Could use the new `Map.of` here..
```
Map<String, Object> metadata = Map.of(
                "1", true,
                "2", 123,
                "3", "string");
```

File: src/main/java/uk/gov/pay/api/json/RequestJsonParser.java [hunk 2]
```diff
@@ -70,9 +78,22 @@ static CreatePaymentRequest parsePaymentRequest(JsonNode paymentRequest) {
...
+        if (paymentRequest.has(METADATA)) {
+            Map<String, Object> metadataMap = null;
+            try {
+                metadataMap = objectMapper.convertValue(paymentRequest.get("metadata"), Map.class);
+            } catch (IllegalArgumentException e) {
+                PaymentError paymentError = new PaymentError(METADATA, CREATE_PAYMENT_METADATA_VALIDATION_ERROR, "Field [metadata] must be an object of JSON key-value pairs");
+                throw new WebApplicationException(Response.status(HttpStatus.SC_UNPROCESSABLE_ENTITY).entity(paymentError).build());
+            }
+            ExternalMetadata metadata = new ExternalMetadata(metadataMap);
+            Set<ConstraintViolation<ExternalMetadata>> violations = validator.validate(metadata);
```
Comment (danworth) [inline src/main/java/uk/gov/pay/api/json/RequestJsonParser.java]: Seems like a sensible way of doing it which is a stepping stone towards using the validator throughout 👍

File: src/test/java/uk/gov/pay/api/it/validation/PaymentResourceMetadataValidationFailuresITest.java [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+        CreateChargeRequestParams createChargeRequestParams = createChargeRequestParamsBuilder
+                .withMetadata(Map.of("foo", List.of("cake", "chocolate"), "bar", Map.of("a", "b")))
+                .build();
+
+        assertMetadataValidationError(createChargeRequestParams, 
+                "Invalid attribute value: Field [metadata] values must be of type String, Boolean or Number");
+    }
+
+    @Test
+    @Parameters({"", " ", "keyMoreThanThirtyCharskeyMoreThanThirtyCharskeyMoreThanThirtyChars"})
```
Comment (danworth) [inline src/test/java/uk/gov/pay/api/it/validation/PaymentResourceMetadataValidationFailuresITest.java]: Maybe show `@Parameters` at a team tech meeting, looks useful but I don't see it being widely used.

File: src/test/java/uk/gov/pay/api/it/validation/PaymentResourceMetadataValidationFailuresITest.java (hunk 1 above)
Comment (oswaldquek) [reply src/test/java/uk/gov/pay/api/it/validation/PaymentResourceMetadataValidationFailuresITest.java]: `Map.of` takes a max of 10 keys :)

File: src/test/java/uk/gov/pay/api/it/validation/PaymentResourceMetadataValidationFailuresITest.java (hunk 3 above)
Comment (oswaldquek) [reply src/test/java/uk/gov/pay/api/it/validation/PaymentResourceMetadataValidationFailuresITest.java]: it's used in connector too!

**Pull request merged**